### ISSUE TITLE
ListItem should stuff value as child content rather than label attribute

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#1145]
@billreed63, @peller

Assigning to @jhpedemonte because @billreed63 is all booked up, but @jhpedemonte might be all booked up, too.

Remember we had issues with one of the first Preview 3 users with ListItem using 'label' attribute instead of child content? But besides the flakiness we saw with 'label' attribute, there is another reason for putting text content as innerHTML instead of within attribute: sometimes users will want to use rich markup. (That is, include some markup characters within SmartInput so that the radio button for "HTML markup" appears.) In fact, a user request is the reason I started poking around in this area. Sometimes users want a 2-line value as text content, where the top line is normal size and the bottom line is a smaller size:

```
MAJOR TEXT
minor text
```

Since dojox.mobile doesn't support major/minor text natively within the widget, the way to pull it off with dojo 1.7 is:
- Enter the following into smart input: `<div>MAJOR TEXT</div><div class='ListItemSubTitle'>minor text</div>`
- In Properties palette, set 'height' on the ListItem to a larger value. (Defaults to 43px, perhaps set to 63px)
- In Properties palette, might want to set 'line-height' to a smaller value. (Defaults to 43px, perhaps set to 38px)
- In app.css, define the ListItemSubTitle class to something like this:

```
.ListItemSubTitle {
  margin-left:1em;
  font-size:12px;
  line-height:12px;
  font-weight:normal;
}
```

This bug is about putting the ListItem child content into innerHTML rather than the 'label' attribute. The reason for that is that while it is possible to put markup in the 'label' attribute, it's really ugly with all of the entities and therefore makes it virtually impossible to enter the desired markup in source view. It's probably just a few minutes to change widgets.json to say stuff the content into innerHTML.

@peller - probably good to ask the dojox.mobile guys about the possibility of supporting multi-line text values in ListItem in a future release so that authors don't have to resort to the monkey business I'm outlining above.

**Comment by peller** [comment]
It's unclear from the request whether fixed item height is acceptable across the list

**Comment by JonFerraiolo** [comment]
@peller, given your comment, could you please ask the dojox.mobile guys about 2 things: (1) do they see any issues with dojox.mobile 1.7 if a user puts multiline text within a ListItem child content and assigns a specific 'height' and 'line-height' to a specific ListItem? (2) Any chance they can support multiple line values in ListItems down the road, where second line has smaller text?

**Comment by peller** [comment]
I've asked Kamiyama-san and referred him to this ticket

**Comment by peller** [comment]
Kamiyama-san pointed out there's a variableHeight property.  Is this the solution?

**Comment by JonFerraiolo** [comment]
Yes, variableHeight does help and is maybe as far as Kamiyama-san wants to go with this. It doesn't deliver the full requirement - subsequent lines should be smaller size - perhaps Kamiyama-san will say that users could do something like this:

```
<style type='text/css'>
.ListItemSubItem { font-size:80%; }
</style>
...
<span dojoType='dojox.mobile.ListItem' variableHeight='true'>
<div>Line 1</div>
<div class='ListItemSubItem'>smaller size text</div>
</span>
```

However, we don't include variableHeight in our metadata at this point, so it doesn't appear in the list of widget-specific properties. I opened this issue: #1179

**Comment by peller** [comment]
Yes, I think that's the right answer.  It's all just markup and CSS,  no need to add more code or restrictions here.  If anything, maybe just some CSS rules as part of the theme for major/minor text in an item?

**Comment by JonFerraiolo** [comment]
Raising to Priority-Med. Not sure why this was given low priority. Seems like a relatively easy and localized fix, is needed for PhoneGap deployment, and addresses a common user requirement.

**Comment by JonFerraiolo** [comment]
See #1253

**Comment by JonFerraiolo** [comment]
Reassigned to me. I have to fix this one to fix #1253. I am making progress on this one.
